python/ql/src/semmle/python/web/turbogears/Response.qll

```diff
@@ -0,0 +1,38 @@
+import python
+
+import semmle.python.security.TaintTracking
+import semmle.python.security.strings.Untrusted
+
+import TurboGears
+
+
+
+class ControllerMethodReturnValue extends TaintSink {
+
+    ControllerMethodReturnValue() {
+        exists(TurboGearsControllerMethod m |
+            m.getAReturnValueFlowNode() = this and
+            not m.isTemplated()
+        )
+    }
+
+    override predicate sinks(TaintKind kind) {
+        kind instanceof ExternalStringKind
```

Comment:
Should this be just `StringKind`?
(And similarly for `ExternalStringDictKind` below.)

Reply:
`StringKind` is just for flow, and doesn't indicate any form of trust (or lack of trust).

Reply:
Maybe I'm just confused, but why would the _response_ be classified as an _external_, untrusted string? This does not match up with what we do for e.g. Flask or Tornado. Also, unless I'm mistaken, it would prevent the stack trace exposure query from working, as `ExceptionInfo` extends `StringKind` and not `ExternalStringKind`.

Reply:
You are right about the `ExceptionInfo` extending `StringKind`, so it does need to be `StringKind`, not `ExternalStringKind`. I was thinking of `XSS` where we are worried about external strings.